FreeRTOS/FreeRTOS-Plus-TCP
# tcp netstat utility, new tcp_sources directory

Description
-----------
A few things happened in this PR:

1) created 2 new directories where non-essential but useful sources can be stored:
~~~
    tools/tcp_sources
    tools/tcp_sources/include
~~~

2) Created a function that retrieves information about sockets, open ports, and some basic I/O statistics:
~~~
    tools/tcp_sources/tcp_netstat.c
    tools/tcp_sources/include/tcp_netstat.h
~~~

3) move some sources code from the tools directory into the new directories:
~~~
    tools/tcp_sources/tcp_mem_stats.c
    tools/tcp_sources/tcp_dump_packets.c
    tools/tcp_sources/tcp_mem_stats.md
    tools/tcp_sources/tcp_dump_packets.md
    tools/tcp_sources/include/tcp_mem_stats.h
    tools/tcp_sources/include/tcp_dump_packets.h
~~~

4) Removed the last few tabs, mostly found in comments.

5) Removed the explicit `#include` of `FreeRTOSIPConfigDefaults.h` in some c modules.

Test Steps
-----------
Make sure that your FreeRTOSIPConfig.h includes tools/tcp_sources/include/tcp_netstat.h:

    `@include "tcp_netstat.h"`

Tun the application for a while, bot UDP and TCP sockets. And then execute the following code:
~~~c
    MetricsType_t xMetrics;
    vGetMetrics( &( xMetrics ) );
    vShowMetrics ( &( xMetrics ) );
~~~
The function `vShowMetrics()` will print a list of all information, by using the `FreeRTOS_printf()` macro.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
